src/model/document/document.ts

```diff
@@ -206,6 +208,45 @@ function useElementFocus() {
   }
 }
 
+function useElementDrag() {
+  function makeDraggable(element: QuantumElement, gridCellSize: Vector2) {
+    // wait until the element gets created in the DOM
+    nextTick(function () {
+      var DOMElement = document.getElementById(element.id)
```

Comment:
While I admit that `domElement` does look slightly odd, I'd still subjectively prefer a lowercase name here. And reserve uppercase names for classes.

Reply:
Also, thinking about it for a bit, it feels odd to be accessing the DOM in here. Maybe this function should be moved to the UI part of the code? As in, keeping the `src/model` stuff mostly independent of how it looks like on the screen and stuff.
(If you want, I can take a closer look at that)

Reply:
You have done an excellent job at organization and creating a great scaffolding for easy scalability, so I will go with your judgement here. I will see about moving it someplace else.

Reply:
I moved it to `QuantumDocument.vue`.  LMK what you think if it here.

Reply:
Definitely a fan of it. I noticed that if you tell interactjs to make all elements with a given class name interactive, it will make elements that get created *after* it interactive as well. Which means that I got to simplify the code a bit more :)